src/client/routes/user_reports.tsx

```diff
@@ -1,14 +1,26 @@
 import { DateToYMDString } from "../helpers/datetime";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import UserReportFrame from "../components/user_report_frame";
 
 export default function UserReports() {
+  const searchParams = new URL(document.location.toString()).searchParams;
+
   // The default selected date is two days ago. We're running on processed
   // telemetry data that could be up to 24 hours outdated, so this makes sure
   // we're not missing reports.
-  const defaultFromDate = new Date();
-  defaultFromDate.setDate(defaultFromDate.getDate() - 2);
+  let defaultFromDate = new Date();
+  if (searchParams.has("from")) {
+    defaultFromDate = new Date(searchParams.get("from")!);
+  } else {
+    defaultFromDate.setDate(defaultFromDate.getDate() - 2);
+  }
+
   const [fromDate, setFromDate] = useState(DateToYMDString(defaultFromDate));
+  useEffect(() => {
+    const targetUrl = new URL(document.location.toString());
+    targetUrl.searchParams.set("from", fromDate);
+    history.replaceState({}, document.title, targetUrl);
```

Comment:
I wonder if we should use "pushState" here? It seems like one might want to navigate back and forward between different dates.

Reply:
`pushState` is weird. You're right of course, it would be better, but the browser doesn't actually reload the page when navigating to a state that has been `pushState`d. So to make that work, I'd have to add a `popstate` listener to the JS and handle navigation there.

... although, when writing this, I realized that `react-router` probably already knows how to do that, and I shouldn't even try to do it myself. I'll look into this and push an update when I'm done.

Reply:
Okay, done. I'm now using the framework-provided routing APIs for the search parameter handling, and everything magically works.